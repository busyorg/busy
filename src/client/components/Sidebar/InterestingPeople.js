import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';
import User from './User';
import './InterestingPeople.less';
import './SidebarContentBlock.less';

const InterestingPeople = ({ users, onRefresh }) => (
  <div className="InterestingPeople SidebarContentBlock">
    <h4 className="SidebarContentBlock__title">
      <i className="iconfont icon-group SidebarContentBlock__icon" />{' '}
      <FormattedMessage id="interesting_people" defaultMessage="Interesting People" />
      <i
        role="presentation"
        onClick={onRefresh}
        className="iconfont icon-refresh InterestingPeople__icon-refresh"
      />
    </h4>
    <div className="SidebarContentBlock__content">
      {users && users.map(user => <User key={user.name} user={user} />)}
      <h4 className="InterestingPeople__more">
        <Link to={'/discover'}>
          <FormattedMessage id="discover_more_people" defaultMessage="Discover More People" />
        </Link>
      </h4>
    </div>
  </div>
);

InterestingPeople.propTypes = {
  users: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string })),
  onRefresh: PropTypes.func,
};

InterestingPeople.defaultProps = {
  users: [],
  onRefresh: () => {},
};

export default InterestingPeople;
