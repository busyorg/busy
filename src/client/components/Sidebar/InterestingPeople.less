@import (reference) "../../styles/custom.less";

.InterestingPeople__icon-refresh {
  font-size: 22px !important;
  float: right;
  margin-top: 5px;
  cursor: pointer;
}

.InterestingPeople__title {
  color: #99aab5;
  font-size: 18px;
  font-weight: 500;
}

.InterestingPeople__more > a {
  color: #99aab5;
  font-weight: 500;
}
