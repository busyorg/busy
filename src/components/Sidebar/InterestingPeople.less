@import (reference) "../../styles/custom.less";

.InterestingPeople {
  background: @white;
  border: 1px solid #e9e7e7;
  border-radius: @border-radius-base;
}

.InterestingPeople__container {
  padding: 8px 16px 16px 16px;
}

.InterestingPeople__icon {
  color: @purple;
  vertical-align: -2px;
}

.InterestingPeople__title {
  color: #99aab5;
  font-size: 18px;
  font-weight: 500;
}

.InterestingPeople__divider {
  width: 100%;
  height: 1px;
  margin: 12px 0;
  background-color: #e9e7e7;
}

.InterestingPeople__more > a {
  color: #99aab5;
  font-weight: 500;
}
