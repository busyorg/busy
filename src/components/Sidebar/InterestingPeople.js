import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';
import User from './User';
import './InterestingPeople.less';

const InterestingPeople = ({ users, onRefresh }) => (
  <div className="InterestingPeople">
    <div className="InterestingPeople__container">
      <h4 className="InterestingPeople__title">
        <i className="iconfont icon-group InterestingPeople__icon" />
        {' '}
        <FormattedMessage id="interesting_people" defaultMessage="Interesting People" />
        <i
          role="presentation"
          onClick={onRefresh}
          className="iconfont icon-refresh InterestingPeople__icon-refresh"
        />
      </h4>
      <div className="InterestingPeople__divider" />
      {users && users.map(user => <User key={user.name} user={user} />)}
      <h4 className="InterestingPeople__more">
        <Link to={'/latest-comments'}>
          <FormattedMessage id="discover_more_people" defaultMessage="Discover More People" />
        </Link>
      </h4>
    </div>
  </div>
);

InterestingPeople.propTypes = {
  users: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string })),
  onRefresh: PropTypes.func,
};

InterestingPeople.defaultProps = {
  users: [],
  onRefresh: () => {},
};

export default InterestingPeople;
