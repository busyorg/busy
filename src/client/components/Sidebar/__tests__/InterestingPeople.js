import React from 'react';
import { shallow } from 'enzyme';
import InterestingPeople from '../InterestingPeople';

describe('<InterestingPeople />', () => {
  it('renders without exploding', () => {
    const props = {
      users: [{ name: 'sekhmet' }, { name: 'fabien' }, { name: 'ekitcho' }, { name: 'jm90mm' }],
    };
    const wrapper = shallow(<InterestingPeople {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
