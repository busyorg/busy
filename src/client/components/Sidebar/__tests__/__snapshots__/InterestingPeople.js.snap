// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InterestingPeople /> renders without exploding 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InterestingPeople
    onRefresh={[Function]}
    users={
      Array [
        Object {
          "name": "sekhmet",
        },
        Object {
          "name": "fabien",
        },
        Object {
          "name": "ekitcho",
        },
        Object {
          "name": "jm90mm",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="SidebarContentBlock__title"
        >
          <i
            className="iconfont icon-group SidebarContentBlock__icon"
          />
           
          <FormattedMessage
            defaultMessage="Interesting People"
            id="interesting_people"
            values={Object {}}
          />
          <i
            className="iconfont icon-refresh InterestingPeople__icon-refresh"
            onClick={[Function]}
            role="presentation"
          />
        </h4>,
        <div
          className="SidebarContentBlock__content"
        >
          <User
            user={
              Object {
                "name": "sekhmet",
              }
            }
          />
          <User
            user={
              Object {
                "name": "fabien",
              }
            }
          />
          <User
            user={
              Object {
                "name": "ekitcho",
              }
            }
          />
          <User
            user={
              Object {
                "name": "jm90mm",
              }
            }
          />
          <h4
            className="InterestingPeople__more"
          >
            <Link
              replace={false}
              to="/discover"
            >
              <FormattedMessage
                defaultMessage="Discover More People"
                id="discover_more_people"
                values={Object {}}
              />
            </Link>
          </h4>
        </div>,
      ],
      "className": "InterestingPeople SidebarContentBlock",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="iconfont icon-group SidebarContentBlock__icon"
            />,
            " ",
            <FormattedMessage
              defaultMessage="Interesting People"
              id="interesting_people"
              values={Object {}}
            />,
            <i
              className="iconfont icon-refresh InterestingPeople__icon-refresh"
              onClick={[Function]}
              role="presentation"
            />,
          ],
          "className": "SidebarContentBlock__title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "iconfont icon-group SidebarContentBlock__icon",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defaultMessage": "Interesting People",
              "id": "interesting_people",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "iconfont icon-refresh InterestingPeople__icon-refresh",
              "onClick": [Function],
              "role": "presentation",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <User
                user={
                  Object {
                    "name": "sekhmet",
                  }
                }
              />,
              <User
                user={
                  Object {
                    "name": "fabien",
                  }
                }
              />,
              <User
                user={
                  Object {
                    "name": "ekitcho",
                  }
                }
              />,
              <User
                user={
                  Object {
                    "name": "jm90mm",
                  }
                }
              />,
            ],
            <h4
              className="InterestingPeople__more"
            >
              <Link
                replace={false}
                to="/discover"
              >
                <FormattedMessage
                  defaultMessage="Discover More People"
                  id="discover_more_people"
                  values={Object {}}
                />
              </Link>
            </h4>,
          ],
          "className": "SidebarContentBlock__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "sekhmet",
            "nodeType": "function",
            "props": Object {
              "user": Object {
                "name": "sekhmet",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "fabien",
            "nodeType": "function",
            "props": Object {
              "user": Object {
                "name": "fabien",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ekitcho",
            "nodeType": "function",
            "props": Object {
              "user": Object {
                "name": "ekitcho",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "jm90mm",
            "nodeType": "function",
            "props": Object {
              "user": Object {
                "name": "jm90mm",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/discover"
              >
                <FormattedMessage
                  defaultMessage="Discover More People"
                  id="discover_more_people"
                  values={Object {}}
                />
              </Link>,
              "className": "InterestingPeople__more",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <FormattedMessage
                  defaultMessage="Discover More People"
                  id="discover_more_people"
                  values={Object {}}
                />,
                "replace": false,
                "to": "/discover",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "Discover More People",
                  "id": "discover_more_people",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="SidebarContentBlock__title"
          >
            <i
              className="iconfont icon-group SidebarContentBlock__icon"
            />
             
            <FormattedMessage
              defaultMessage="Interesting People"
              id="interesting_people"
              values={Object {}}
            />
            <i
              className="iconfont icon-refresh InterestingPeople__icon-refresh"
              onClick={[Function]}
              role="presentation"
            />
          </h4>,
          <div
            className="SidebarContentBlock__content"
          >
            <User
              user={
                Object {
                  "name": "sekhmet",
                }
              }
            />
            <User
              user={
                Object {
                  "name": "fabien",
                }
              }
            />
            <User
              user={
                Object {
                  "name": "ekitcho",
                }
              }
            />
            <User
              user={
                Object {
                  "name": "jm90mm",
                }
              }
            />
            <h4
              className="InterestingPeople__more"
            >
              <Link
                replace={false}
                to="/discover"
              >
                <FormattedMessage
                  defaultMessage="Discover More People"
                  id="discover_more_people"
                  values={Object {}}
                />
              </Link>
            </h4>
          </div>,
        ],
        "className": "InterestingPeople SidebarContentBlock",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="iconfont icon-group SidebarContentBlock__icon"
              />,
              " ",
              <FormattedMessage
                defaultMessage="Interesting People"
                id="interesting_people"
                values={Object {}}
              />,
              <i
                className="iconfont icon-refresh InterestingPeople__icon-refresh"
                onClick={[Function]}
                role="presentation"
              />,
            ],
            "className": "SidebarContentBlock__title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "iconfont icon-group SidebarContentBlock__icon",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "Interesting People",
                "id": "interesting_people",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "iconfont icon-refresh InterestingPeople__icon-refresh",
                "onClick": [Function],
                "role": "presentation",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <User
                  user={
                    Object {
                      "name": "sekhmet",
                    }
                  }
                />,
                <User
                  user={
                    Object {
                      "name": "fabien",
                    }
                  }
                />,
                <User
                  user={
                    Object {
                      "name": "ekitcho",
                    }
                  }
                />,
                <User
                  user={
                    Object {
                      "name": "jm90mm",
                    }
                  }
                />,
              ],
              <h4
                className="InterestingPeople__more"
              >
                <Link
                  replace={false}
                  to="/discover"
                >
                  <FormattedMessage
                    defaultMessage="Discover More People"
                    id="discover_more_people"
                    values={Object {}}
                  />
                </Link>
              </h4>,
            ],
            "className": "SidebarContentBlock__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "sekhmet",
              "nodeType": "function",
              "props": Object {
                "user": Object {
                  "name": "sekhmet",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "fabien",
              "nodeType": "function",
              "props": Object {
                "user": Object {
                  "name": "fabien",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ekitcho",
              "nodeType": "function",
              "props": Object {
                "user": Object {
                  "name": "ekitcho",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "jm90mm",
              "nodeType": "function",
              "props": Object {
                "user": Object {
                  "name": "jm90mm",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/discover"
                >
                  <FormattedMessage
                    defaultMessage="Discover More People"
                    id="discover_more_people"
                    values={Object {}}
                  />
                </Link>,
                "className": "InterestingPeople__more",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <FormattedMessage
                    defaultMessage="Discover More People"
                    id="discover_more_people"
                    values={Object {}}
                  />,
                  "replace": false,
                  "to": "/discover",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "defaultMessage": "Discover More People",
                    "id": "discover_more_people",
                    "values": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
